### DISCUSSION TITLE
TerminalGuiDesigner is awesome

### DISCUSSION THREAD

**Discussion by BoldInventions** [tznind/TerminalGuiDesigner#232] [General]
Thanks for making this.

It is a little buggy, but 1000% better than doing it by hand with code.

**Comment by tznind** [comment]
Thanks for taking the time to leave this post.  Getting feedback from users is very nice and highly motivating 👍 .

Have you had any hard crashes? or lost any work as a result of bugs? These are the most critical to prevent.

But I am interested in the other buggy behaviour too (visual, commands not working etc).  If you have the time to describe reproducible steps to demonstrate the bugs, that would be really helpful.

**Reply by BoldInventions** [reply]
No crashes.  No lost work.  :)

Since we can make .NET apps on Linux, but aren't allowed to use WinForms, I think Terminal Gui is really useful, and your designer tool saved me probably 20 to 40 hours of working trying to do it programmatically.   

[https://files.catbox.moe/ux6naq.png](https://files.catbox.moe/ux6naq.png)   <-- Here's what I'm working on.

I think most of my problems are that I'm using it without learning the right way to do things.

I'm using the mouse and that's the part that barely works on combo boxes.   You have to click on the little arrow on the right, otherwise you can't move the control.  

If a label is small  (1x1) the mouse will only resize - you can't set the position using the mouse, so I had to set the xy pos manually.

I had problems when trying to select items in frameviews with the mouse.  

The mouse barely works at all on combo boxes.

It would be really helpful if you could change the cursor depending on if it is going resize or move the control.  Maybe that isn't possible but that is the main problem I had.  That and being able to move my 1x1 labels with the mouse would be good.

Also, if you right click on a control in a frameview, you should let the user choose which control in tree they want to change, I.E. the frame view or the label.

Also, I couldn't figure out how to add more top-level menu bar items so I had to do that by editing myView.Designer.cs directly.

Anyway, I'm very happy with the designer whether these are fixed or not.   Great job.

**Reply by tznind** [reply]
Thanks for the detailed feedback.  Should be possible to address some of these issues :).  Looks like a cool app you are developing.

**Comment by dodexahedron** [comment]
It sure is!

@tznind has saved a lot of us a lot of time with this.

Once Terminal.Gui v2 is stable, this will get even better. Exciting things are on the way, across the board!
